content/helpers.go

```diff
@@ -37,10 +37,16 @@ var bufPool = sync.Pool{
 	},
 }
 
+type reader interface {
+	Reader() io.Reader
+}
+
 // NewReader returns a io.Reader from a ReaderAt
 func NewReader(ra ReaderAt) io.Reader {
-	rd := io.NewSectionReader(ra, 0, ra.Size())
-	return rd
+	if rd, ok := ra.(reader); ok {
```

Comment:
```suggestion
	if rd, ok := ra.(io.Reader); ok {
```

Reply:
`io.Reader` doesn't help here since the instance cannot be type asserted to `*os.File`. It needs a new reader instance and ideally carrying the size data, which `*LimitedReader` does.